temperature_regulator/project/test/test_txt_config_loader.py

```diff
@@ -0,0 +1,57 @@
+import unittest
+import tempfile
+from main.config_loader.txt_config_loader import TxtConfigLoader
+
+
+class TxtConfigLoaderTestCase(unittest.TestCase):
+    def setUp(self):
+        self._file = tempfile.NamedTemporaryFile(mode="w+t")
+
+    def tearDown(self):
+        self._file.close()
+
+    def add_data_to_file(self, data: str):
+        self._file.write(data)
+        self._file.seek(0)
+
+    def test_listenerSocketAddress_EmptyFile_ThrowsKeyError(self):
+        data = ""
```

Comment:
to już do twojej opinii, ale u nas w allegro w projekcie i ogólnie np. w groovy spock wydziela się takie sekcie: given, when, then. Wtedy mamy trochę (moim zdaniem) czytelniejszy kod bo wiemy jaki jest stan (given), co robimy (when), i czego oczekujemy (then). W pythonie nie ma takich słów kluczowych, ale można dawać np. komentarze (tak robimy w Typescripcie). Ten test mógłby wygladać tak
```python
# given
data = ""
self.add_data_to_file(data)

# when
config = TxtConfigLoader(self._file)

# then
self.assertRaises(KeyError, lambda: config.listener_socket_address)
```
ale to do twojej własnej oceny co jest czytelniejsze